app/presenters/view-licences/view-licence.presenter.js

```diff
@@ -0,0 +1,38 @@
+'use strict'
+const { format, isAfter } = require('date-fns')
+const { dataFormat } = require('../../lib/static-lookups.lib.js')
+
+/**
+ * Formats the end date of the licence
+ * @module ViewLicencePresenter
+ */
+function _endDate (licenceData) {
+  if (licenceData.licence.expiredDate) {
+    const expiredDate = new Date(licenceData.licence.expiredDate)
+    if (!isAfter(Date.now(), expiredDate)) {
+      return `${format(expiredDate, dataFormat)}`
+    }
+    return null
+  }
+  return null
+}
+
+/**
+ * Formats data for the `/view-licence/{id}/` page
+ * @module ViewLicencePresenter
+ */
+function go (licenceData) {
+  const endDate = _endDate(licenceData)
+
+  return {
+    id: licenceData.licence.id,
+    licenceRef: licenceData.licence.licenceRef,
+    region: licenceData.licence.region.displayName,
+    startDate: format(new Date(licenceData.licence.startDate), dataFormat),
+    ...(endDate && { endDate })
+  }
+}
```

Comment:
Just re-reading [function naming conventions for services](https://github.com/DEFRA/water-abstraction-team/blob/main/coding_conventions.md#function-naming-conventions-for-services) we don't explicitly state that public methods go before private, and _then_ in alphabetical order! 🤦 

But that's the main change needed here.

Next up is the comments. We try and give public methods (which is just `go()` 99% of the time) the full [JSDoc treatment](https://jsdoc.app/). On private methods it is 'dealers choice'. If something is complex or exceptional we'll often add a comment. But we don't worry about following JSDoc completely. The suggestion provides an example of how this might look in this case.

> The suggestion also assumes the comment about just returning the `LicenceModel` instance rather than wrapping it as an object in `FetchLicenceService` has been applied.

The dates on the licence have been wrapped in `new Date()` calls. Only because of the [rabbit hole I fell down I believe this is not needed](https://github.com/DEFRA/water-abstraction-system/pull/655#issue-2076278837). So, I've amended that in my suggestion but you may have encountered something different. 🤷 

```suggestion
/**
 * Formats data for the view `/licences/{id}/` page
 *
 * @param {module:LicenceModel} licence - The licence where the data will be extracted for from
 *
 * @returns {Object} The data formatted for the view template
 */
function go (licence) {
  const { expiredDate, id, licenceRef, region, startDate } = licence
  const endDate = _endDate(licence)

  return {
    id,
    endDate: _endDate(expiredDate),
    licenceRef,
    region: region.displayName,
    startDate: formatLongDate(startDate)
  }
}

function _endDate (expiredDate) {
  if (!expiredDate) {
    return null
  }
  
  return formatLongDate(expiredDate)
}
```